pkg/lookups/lookups.go

```
@@ -0,0 +1,153 @@
+	log "github.com/sirupsen/logrus"
+	"github.com/valyala/fasthttp"
+)
+
+const (
+	lookupsDir      = "lookups"
+	allowedExtCSV   = ".csv"
+	allowedExtCSVGZ = ".csv.gz"
+)
+
+func UploadLookupFile(ctx *fasthttp.RequestCtx) {
+	fileName := string(ctx.FormValue("name"))
+	if fileName == "" {
+		log.Error("UploadLookupFile: File name is required")
+		ctx.Error("File name is required", fasthttp.StatusBadRequest)
+		return
+	}
+
+	fileHeader, err := ctx.FormFile("file")
+	if err != nil {
+		log.Errorf("UploadLookupFile: Error retrieving the file: %v", err)
+		ctx.Error("Error retrieving the file", fasthttp.StatusBadRequest)
+		return
+	}
+
+	// Check for .csv and .csv.gz extensions
+	lowerFileName := strings.ToLower(fileHeader.Filename)
+	isCSV := strings.HasSuffix(lowerFileName, allowedExtCSV)
+	isCSVGZ := strings.HasSuffix(lowerFileName, allowedExtCSVGZ)
+
+	if !isCSV && !isCSVGZ {
+		log.Errorf("UploadLookupFile: Invalid file type: %s", filepath.Ext(fileHeader.Filename))
+		ctx.Error(fmt.Sprintf("Invalid file type. Only %s and %s files are allowed", allowedExtCSV, allowedExtCSVGZ), fasthttp.StatusBadRequest)
+		return
+	}
+
+	if !strings.HasSuffix(strings.ToLower(fileName), allowedExtCSV) &&
+		!strings.HasSuffix(strings.ToLower(fileName), allowedExtCSVGZ) {
+		if isCSVGZ {
+			fileName += allowedExtCSVGZ
+		} else {
+			fileName += allowedExtCSV
+		}
+	}
+
+	fullLookupsDir := filepath.Join(config.GetDataPath(), lookupsDir)
+	if err := os.MkdirAll(fullLookupsDir, os.ModePerm); err != nil {
+		log.Errorf("UploadLookupFile: Error creating lookups directory: %v", err)
+		ctx.Error("Error creating lookups directory", fasthttp.StatusInternalServerError)
+		return
+	}
+
+	dstPath := filepath.Join(fullLookupsDir, fileName)
+	if _, err := os.Stat(dstPath); !os.IsNotExist(err) {
+		log.Errorf("UploadLookupFile: File already exists: %s", fileName)
+		ctx.Error("A file with the same name already exists", fasthttp.StatusConflict)
+		return
+	}
```

Do we want to throw an error here always, or provide a way to write over the existing file? If we can't write over a file, they'd have to manually delete the file and then upload the new version which seems a little annoying. But if we do want to add a way to overwrite a file, I think we could do that in a separate small PR if you want

---

I'll address this in a new PR.